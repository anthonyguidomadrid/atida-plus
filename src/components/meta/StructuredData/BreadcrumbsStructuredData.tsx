import { FunctionComponent } from 'react'
import { BreadcrumbList } from 'schema-dts'
import { jsonLdScriptProps } from 'react-schemaorg'
import Head from 'next/head'
import getConfig from 'next/config'
import { useRouter } from 'next/router'

type BreadcrumbsProps = {
  breadcrumbs: {
    title?: string
    slug?: string
  }[]
}

export const BreadcrumbsStructuredData: FunctionComponent<BreadcrumbsProps> = ({
  breadcrumbs
}) => {
  const { publicRuntimeConfig } = getConfig()
  const host = publicRuntimeConfig.host
  const { locale } = useRouter()

  return (
    <Head>
      <script
        data-testid="structured-data-breadcrumbs"
        type="application/ld+json"
        {...jsonLdScriptProps<BreadcrumbList>({
          '@context': 'https://schema.org',
          '@type': 'BreadcrumbList',
          itemListElement: breadcrumbs
            .filter(({ title }) => title)
            .map(({ title, slug }, index) => {
              return {
                '@type': 'ListItem',
                position: index + 1,
                name: title,
                item: slug && `${host}/${locale}/${slug}`
              }
            })
        })}
      />
    </Head>
  )
}
