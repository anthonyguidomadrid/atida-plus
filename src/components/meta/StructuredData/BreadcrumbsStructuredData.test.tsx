import { renderWithStore } from '~test-helpers'
import { BreadcrumbsStructuredData } from './BreadcrumbsStructuredData'

describe(BreadcrumbsStructuredData, () => {
  it('renders first level breadcrumb structured data script tag with json-ld data', () => {
    const { container } = renderWithStore(
      <BreadcrumbsStructuredData
        breadcrumbs={[
          {
            title: 'Rosto'
          }
        ]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders second level breadcrumb structured data script tag with json-ld data', () => {
    const { container } = renderWithStore(
      <BreadcrumbsStructuredData
        breadcrumbs={[
          {
            title: 'Beleza',
            slug: 'beleza'
          },
          {
            title: 'Rosto'
          }
        ]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders third level breadcrumb structured data script tag with json-ld data', () => {
    const { container } = renderWithStore(
      <BreadcrumbsStructuredData
        breadcrumbs={[
          {
            title: 'Beleza',
            slug: 'beleza'
          },
          {
            title: 'Rosto',
            slug: 'beleza/rosto'
          },
          {
            title: 'Acne'
          }
        ]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
