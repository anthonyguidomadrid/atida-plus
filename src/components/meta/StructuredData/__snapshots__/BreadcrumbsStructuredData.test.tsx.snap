// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbsStructuredData renders first level breadcrumb structured data script tag with json-ld data 1`] = `
<script
  data-testid="structured-data-breadcrumbs"
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Rosto"}]}
</script>
`;

exports[`BreadcrumbsStructuredData renders second level breadcrumb structured data script tag with json-ld data 1`] = `
<script
  data-testid="structured-data-breadcrumbs"
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Beleza","item":"somehost.com/en-gb/beleza"},{"@type":"ListItem","position":2,"name":"Rosto"}]}
</script>
`;

exports[`BreadcrumbsStructuredData renders third level breadcrumb structured data script tag with json-ld data 1`] = `
<script
  data-testid="structured-data-breadcrumbs"
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Beleza","item":"somehost.com/en-gb/beleza"},{"@type":"ListItem","position":2,"name":"Rosto","item":"somehost.com/en-gb/beleza/rosto"},{"@type":"ListItem","position":3,"name":"Acne"}]}
</script>
`;
